bundle-workflow/tests/test_integ_workflow/integ_test/test_integ_test_suite.py

```diff
@@ -0,0 +1,169 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import unittest
+from unittest.mock import call, patch
+
+from git.git_repository import GitRepository
+from manifests.build_manifest import BuildManifest
+from manifests.bundle_manifest import BundleManifest
+from manifests.test_manifest import TestManifest
+from test_workflow.integ_test.integ_test_suite import (DependencyInstaller,
+                                                       IntegTestSuite,
+                                                       InvalidTestConfigError,
+                                                       MissingArtifactError,
+                                                       ScriptFinder)
+
+
+@patch("os.makedirs")
+@patch("os.chdir")
+@patch.object(GitRepository, "__checkout__")
+class TestIntegSuite(unittest.TestCase):
+    def setUp(self):
+        os.chdir(os.path.dirname(__file__))
+        with open("data/bundle_manifest.yml", "r") as file:
+            self.bundle_manifest = BundleManifest.from_file(file)
+        with open("data/build_manifest.yml", "r") as file:
+            self.build_manifest = BuildManifest.from_file(file)
+        with open("data/test_manifest.yml", "r") as file:
+            self.test_manifest = TestManifest.from_file(file)
+
+    @patch.object(DependencyInstaller, "install_build_dependencies")
+    @patch("os.path.exists", return_value=True)
+    @patch.object(ScriptFinder, "find_integ_test_script")
+    @patch("test_workflow.integ_test.integ_test_suite.execute")
+    @patch("test_workflow.integ_test.integ_test_suite.LocalTestCluster")
+    def test_execute_with_multiple_test_configs(
+        self, mock_local_test_cluster, mock_system_execute, mock_script_finder, *mock
+    ):
+        test_config, component = self.__get_test_config_and_bundle_component(
+            "job-scheduler"
+        )
+        integ_test_suite = IntegTestSuite(
+            component,
+            test_config,
+            self.bundle_manifest,
+            self.build_manifest,
+            "/tmpdir",
+            "s3_bucket_name",
+        )
+        mock_system_execute.return_value = 200, "success", "failure"
+        mock_local_test_cluster.create().__enter__.return_value = "localhost", "9200"
+        mock_script_finder.return_value = "integtest.sh"
```

Comment:
Do we need `integtest.sh` now since we have integrated `DependencyInstaller` now? If not, can we remove the file

Reply:
The purpose for both is different. DependencyInstaller is meant for installing maven and build dependencies, while `integtest.sh` tool contains the logic to run integration test.

Reply:
Sorry about the confusion. I mentioned the wrong shell script. I meant `integtest_dependencies_opensearch.sh`

Reply:
yeah, that's removed now. https://github.com/opensearch-project/opensearch-build/pull/549#discussion_r713595389